actions/test/run_local.ts

```diff
@@ -55,14 +60,92 @@ const main = async () => {
         console.error("[run_local] Error in processBlock", error);
       }
     });
+  } else if (contractAddressEnv) {
+    // If no blockNumberEnv is provided, then we rebuild the state from the default deployment block
+    if (!blockNumberEnv) {
+      console.log(`[run_rebuild] No block number provided, using default deployment block`);
+    } else {
+      assert(!isNaN(Number(blockNumberEnv)), "blockNumber must be a number");
+    }
+    let fromBlock = blockNumberEnv ? Number(blockNumberEnv) : DEFAULT_DEPLOYMENT_BLOCK;
+
+    // Rebuild the state
+    console.log(`[run_rebuild] Rebuild the state of the conditional orders from the historical events.`);
+    const contractAddress = process.env.CONTRACT_ADDRESS;
+    assert(contractAddress && ethers.utils.isAddress(contractAddress), "contract address is required");
+    const pageSize = process.env.PAGE_SIZE ? parseInt(process.env.PAGE_SIZE) : DEFAULT_PAGE_SIZE;
+
+    // 1. Record the current block number - useful with paging
+    let currentBlockNumber = await provider.getBlockNumber();
+    console.log(`[run_rebuild] Current block number: ${currentBlockNumber}`);
+
+    // 2. Connect to the contract instance
+    const contract = ComposableCoW__factory.connect(contractAddress, provider);
+    
+    // 3. Define the filter.
+    const filter = contract.filters.ConditionalOrderCreated();
+
+    // 4. Get the historical events
+    const replayPlan: ReplayPlan = {}
+    let toBlock;
+    do {
```

Comment:
Because during the replay of the blocks, the chain advances, in which case we will have to modify the `for` condition as the loop advances (the chain is continually moving as it's syncing, so it needs to "converge" on the latest).

Reply:
But will this then run forever? 

i still think this could use a more functional approach. Anyways, not big deal, we can refactor later if we feel for it